src/ex4/server/services/utils.js

```diff
@@ -0,0 +1,35 @@
+import { promises as fs } from "fs";
+import { dirname, join } from "path";
+import { fileURLToPath } from "url";
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+const tasksFile = join(__dirname, "..", "data", "tasks.json");
+
+async function readTasksFile() {
```

Comment:
you need to verify if the file exists, both for reading and writing as it failed my local env. you can check if you should create It on server load